specs/capri/thin_opensds_design.md

```diff
@@ -0,0 +1,42 @@
+# Thin OpenSDS Design
+
+**Authors:** [Xing Yang](https://github.com/xing-yang)
+
+This document serves as a design proposal for Thin OpenSDS. This proposal was drafted [here](https://docs.google.com/document/d/16zLCE1X8bNjkGoKvBpqDZaxEHZapWluQVgdvlxuR_tE/edit?usp=sharing).
+
+## Background
+
+In containerized environment, users may not need the full-blown orchestration capabilities of OpenSDS.  Instead, they just want a CSI plugin that can provision storage for Container Orchestration Systems (COs).  That’s why we are proposing to provide a light-weight version of OpenSDS that can serve this need.
+
+## Objectives
+
+### Goals
+
+* Provide a light-weight version of OpenSDS that includes a CSI plugin to provision storage for COs.
+* Only one storage backend is supported at a given time based on configure setting.
+* Only functionalities supported by CSI spec [1] are supported.  These include create/delete volume, attach/detach volume, mount/unmount volume, create/delete snapshot, resize volume, etc.
```

Comment:
Do we have some plan for enabling create/delete fileshare, create/delete fileshare snapshot for container storage?

Reply:
Those will be supported by the OpenSDS CSI plugin but there won't be separate CSI PRCs for them.  They are all called volume interfaces, whether underneath it is supported by block store or shared filesystems.